### ISSUE TITLE
SetErrorHandler should support param with signature

### ISSUE THREAD

**Issue by Lingerhk** [RichardKnop/machinery#496]
Just like the source code in exaple：
	errorhandler := func(err error) {
		log.ERROR.Println("I am an error handler:", err)
	}
	pretaskhandler := func(signature *tasks.Signature) {
		log.INFO.Println("I am a start of task handler for:", signature.Name)
	}
	posttaskhandler := func(signature *tasks.Signature) {
		log.INFO.Println("I am an end of task handler for:", signature.Name)
	}
	worker.SetPostTaskHandler(posttaskhandler)
	worker.SetErrorHandler(errorhandler)
	worker.SetPreTaskHandler(pretaskhandler)
--------------------------------------------------------------------------
I think the "errorhandler" also need param "signature *tasks.Signature", like the follow：
	errorhandler := func(signature *tasks.Signature，err error) {
		log.ERROR.Printf("I am an error handler(task:%s):", signature. Name, err)
	}

**Comment by subsociety** [comment]
I agree with @Lingerhk.  I'd like to be able to have access to the signature so that I can log or use signature data as part of my error handling.

**Comment by chesson1017** [comment]
If errors, posttaskhandler will also be excuted. So u can use pre and post handler only.

**Comment by jaskirat-s** [comment]
@RichardKnop Can I take it up and submit a PR? Can this change be accepted?

**Comment by jaskirat-s** [comment]
> If errors, posttaskhandler will also be excuted. So u can use pre and post handler only.

That's true, but it would be called for every task. For error cases you might want to push the task into a deadset for manual inspection/retry it's prohibitive right now because the error handler is not passed the task.

**Comment by kushalhalder** [comment]
> > If errors, posttaskhandler will also be excuted. So u can use pre and post handler only.
> 
> That's true, but it would be called for every task. For error cases you might want to do error specific handling (like push the task into a deadset for manual inspection/retry) and it's prohibitive right now because the error handler is not passed the task; it's limited to error logging which is not very helpful.

For this use case alone, you can use `onError`
